### PR TITLE
Fix Usability issue with Macros and add new feature to Macros (RM #6969) #4735

### PR DESCRIPTION
**Pull request by RohitBhati8269**
More Details :- [task-link](https://github.com/pgadmin-org/pgadmin4/issues/4735)

### CODE REVIEW

File: web/pgadmin/model/__init__.py [hunk 1]
```diff
@@ -424,20 +424,21 @@ class Macros(db.Model):
...
     __tablename__ = 'macros'
     id = db.Column(db.Integer, primary_key=True)
-    alt = db.Column(db.Boolean(), nullable=False)
-    control = db.Column(db.Boolean(), nullable=False)
-    key = db.Column(db.String(32), nullable=False)
-    key_code = db.Column(db.Integer, nullable=False)
+    alt = db.Column(db.Boolean(), nullable=True)
+    control = db.Column(db.Boolean(), nullable=True)
+    key = db.Column(db.String(32), nullable=True)
+    key_code = db.Column(db.Integer, nullable=True)
```
Comment (akshay-joshi) [inline web/pgadmin/model/__init__.py:430]: This change is not required.

File: web/pgadmin/tools/sqleditor/__init__.py [hunk 2]
```diff
@@ -2692,3 +2693,15 @@ def update_macros(trans_id):
...
+
+
+@blueprint.route(
+    '/get_user_macros',
+    methods=["GET"], endpoint='get_user_macros'
+)
+@pga_login_required
+def user_macros(json_resp=True):
+    """
+    This method is used to get all the columns for data sorting dialog.
```
Comment (akshay-joshi) [inline web/pgadmin/tools/sqleditor/__init__.py:2705]: Correct the function comment. Seems copy paste error.

File: web/pgadmin/tools/sqleditor/static/js/components/dialogs/MacrosDialog.jsx [hunk 3]
```diff
@@ -89,59 +115,87 @@ const useStyles = makeStyles((theme)=>({
...
+function getRandomName(existingDataCount, existingNames) {
+  let newName = `Macro ${existingDataCount + 1}`;
+  // Check if the new name already exists then generating new name.
+  while (existingNames.includes(newName)) {
+    existingDataCount++;
+    newName = `Macro ${existingDataCount + 1}`;
+  }
+  return newName;
+}
+
```
Comment (akshay-joshi) [inline web/pgadmin/tools/sqleditor/static/js/components/dialogs/MacrosDialog.jsx:154]: A similar function is defined in the QueryToolComponent.jsx. Please make a common function and use it.

**Review by akshay-joshi — CHANGES_REQUESTED** [review]
1. If key binding is not there, then no need to display empty brackets.
2. When you add a new row focus should be on SQL control.
3. Default macro names should be picked using the id of the macros table.

**Comment by RohitBhati8269** [comment]
> * When you add a new row focus should be on SQL control.

@akshay-joshi  As just this one is not updated as per our discussion on scrum.

File: web/pgadmin/model/__init__.py [hunk 4]
```diff
@@ -424,20 +424,21 @@ class Macros(db.Model):
...
     __tablename__ = 'macros'
     id = db.Column(db.Integer, primary_key=True)
-    alt = db.Column(db.Boolean(), nullable=False)
-    control = db.Column(db.Boolean(), nullable=False)
-    key = db.Column(db.String(32), nullable=False)
-    key_code = db.Column(db.Integer, nullable=False)
+    alt = db.Column(db.Boolean())
+    control = db.Column(db.Boolean())
+    key = db.Column(db.String(32))
+    key_code = db.Column(db.Integer)
```
Comment (akshay-joshi) [inline web/pgadmin/model/__init__.py:430]: Revert this to the original logic. No changes needed.

**Review by akshay-joshi — CHANGES_REQUESTED** [review]

**Pull request merged**